### PR TITLE
-REMOVE_SF with name

### PR DESCRIPTION
**Pull request by p-leroy**
Possibility to put the name of the scalar field to remove when using the command line option -REMOVE_SF
-REMOVE_SF {SF index or 'SF name'}

### CODE REVIEW

File: qCC/ccCommandLineCommands.cpp [hunk 1]
```diff
@@ -2584,31 +2584,61 @@ bool CommandRemoveSF::process(ccCommandLineInterface &cmd)
...
-
-	if (sfIndex < 0)
+	else
 	{
-		return cmd.error(QObject::tr("Invalid SF index (positive value expected)"));
+		sfIndex = sfIndexStr.toInt(&validSFIndex);
+		if (!validSFIndex)
+		{
+			sfIndex = -1;
+			if (sfIndexStr.size() >= 2 && sfIndexStr.startsWith('\''))
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:2604]: maybe we should move this code in a kind of 'helper function' and reuse it elsewhere? (I believe other methods offer this possibility to deduce the SF from its name now).

File: libs/qCC_io/cmake/GDALSupport.cmake [hunk 2]
```diff
@@ -42,7 +42,7 @@ function( target_link_GDAL ) # ARGV0 = project name
                         ${GDAL_BIN_DIR}/netcdf.dll
                         ${GDAL_BIN_DIR}/geotiff.dll
                         ${GDAL_BIN_DIR}/tiff.dll
-                        ${GDAL_BIN_DIR}/proj_8_0.dll
```
[the remark this answers is not in the document]
Comment (p-leroy) [reply libs/qCC_io/cmake/GDALSupport.cmake:45]: Sure, sorry for that.

File: qCC/ccCommandLineCommands.cpp (hunk 1 above)
Comment (p-leroy) [reply qCC/ccCommandLineCommands.cpp:2604]: Good suggestion, I will have a closer look to it. Thanks for the feedback!

File: qCC/ccCommandLineCommands.cpp [hunk 3]
```diff
@@ -165,6 +165,50 @@ constexpr char OPTION_FILE_NAMES[]						= "FILE";
 constexpr char OPTION_ORIENT[]							= "ORIENT";
 constexpr char OPTION_MODEL[]							= "MODEL";
 
+bool get_SFIndex_SFName(ccCommandLineInterface& cmd, int& sfIndex, QString& sfName)
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:168]: To remain consistent with the project coding style, I would name it:
_static bool GetSFIndexOrName(...)_

File: qCC/ccCommandLineCommands.cpp [hunk 4]
```diff
@@ -165,6 +165,50 @@ constexpr char OPTION_FILE_NAMES[]						= "FILE";
...
+{
+	sfIndex = -1;
+	sfName = "";
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:171]: sfName.clear();

File: qCC/ccCommandLineCommands.cpp [hunk 5]
```diff
@@ -165,6 +165,50 @@ constexpr char OPTION_FILE_NAMES[]						= "FILE";
...
+	bool validSFIndexOrName = true;
+	sfName = cmd.arguments().takeFirst();
+	if (sfName.toUpper() == OPTION_LAST)
+	{
+		sfIndex = -2;
+		cmd.print(QObject::tr("Set active SF index: LAST"));
+	}
+	else
+	{
+		bool validInt;
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:181]: bool validInt = false;

File: qCC/ccCommandLineCommands.cpp [hunk 6]
```diff
@@ -165,6 +165,50 @@ constexpr char OPTION_FILE_NAMES[]						= "FILE";
...
+				}
+				else
+				{
+					sfName = sfName.mid(1, sfName.size() - 2);
+					cmd.print(QObject::tr("Set active SF name: '%1'").arg(sfName));
+					validSFIndexOrName = true;
+				}
+			}
+		}
+		cmd.print(QObject::tr("Set active SF index: %1").arg(sfIndex));
```
Comment (dgirardeau) [inline qCC/ccCommandLineCommands.cpp:206]: Maybe change the message depending on whether it's an index or a name?

File: qCC/ccCommandLineCommands.cpp (hunk 1 above)
Comment (p-leroy) [reply qCC/ccCommandLineCommands.cpp]: I have added a generic function to parse the SF index / name

File: libs/qCC_io/cmake/GDALSupport.cmake (hunk 2 above)
[the remark this answers is not in the document]
Comment (p-leroy) [reply libs/qCC_io/cmake/GDALSupport.cmake]: I have restored the master revision of GDALSupport.cmake.

File: qCC/ccCommandLineCommands.cpp (hunk 6 above)
Comment (p-leroy) [reply qCC/ccCommandLineCommands.cpp:206]: ok

**Review by dgirardeau — APPROVED** [review]

**Pull request merged**